packages/puppeteer-core/src/node/ScreenRecorder.ts

```diff
@@ -56,25 +62,73 @@ export class ScreenRecorder extends PassThrough {
   #controller = new AbortController();
   #lastFrame: Promise<readonly [Buffer, number]>;
 
+  #fps: number;
+
   /**
    * @internal
    */
   constructor(
     page: Page,
     width: number,
     height: number,
-    {speed, scale, crop, format, path}: ScreenRecorderOptions = {},
+    {
+      speed,
+      scale,
+      crop,
+      format,
+      fps,
+      loop,
+      delay,
+      quality,
+      colors,
+      path,
+    }: ScreenRecorderOptions = {},
   ) {
     super({allowHalfOpen: false});
 
+    format ??= 'webm';
+    fps ??= DEFAULT_FPS;
+    loop ||= -1;
+    delay ??= -1;
+    quality ??= CRF_VALUE;
+    colors ??= 256;
     path ??= 'ffmpeg';
 
+    this.#fps = fps;
+
     // Tests if `ffmpeg` exists.
     const {error} = spawnSync(path);
     if (error) {
       throw error;
     }
 
+    const filters = [
+      `crop='min(${width},iw):min(${height},ih):0:0'`,
+      `pad=${width}:${height}:0:0`,
+    ];
+    if (speed) {
+      filters.push(`setpts=${1 / speed}*PTS`);
+    }
+    if (crop) {
+      filters.push(`crop=${crop.width}:${crop.height}:${crop.x}:${crop.y}`);
+    }
+    if (scale) {
+      filters.push(`scale=iw*${scale}:-1:flags=lanczos`);
+    }
+
+    const formatArgs = this.#getFormatArgs(
+      format,
+      fps,
+      loop,
+      delay,
+      quality,
+      colors,
+    );
+    const vf = formatArgs.indexOf('-vf');
+    if (~vf) {
```

Comment:
Let's just check `if (vf !== -1)` it easier to understand.
Also in the next line if we push a empty string won't we end up with `<previous-filter>,,`
Which could break the command line in theory?

Reply:
> Also in the next line if we push a empty string won't we end up with <previous-filter>,,
Which could break the command line in theory?

@Lightning00Blade Well I tested all 3 output formats, and nothing broke. But I agree it's not ideal… Also, this was purely to stop TypeScript complaining about possible `undefined`!